onnxruntime/test/providers/dnnl/math/quantize_linear_ops_test.cc

```diff
@@ -0,0 +1,63 @@
+#include "gtest/gtest.h"
+#include "test/providers/provider_test_utils.h"
+
+
+namespace onnxruntime {
+namespace test {
+
+#ifdef USE_DNNL
+// The same as the default provider, but in this case with constant initializers to test optimization
```

Comment:
these are the same tests as https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/test/providers/cpu/tensor/quantize_linear_test.cc right?
those would already be executed with DNNL EP when enabled so then we don't need to repeat them here?

Reply:
They are almost the same, the only difference is that in these new tests the zero point input has the constant initializer flag set to true, this can be seen [here](https://github.com/eralmual/onnxruntime/blob/f6370755fe56e03c07997c238b488047a84a7d0a/onnxruntime/test/providers/dnnl/math/quantize_linear_ops_test.cc#L16), while the [original tests](https://github.com/microsoft/onnxruntime/blob/eb6aa861cfa7295ee9f7145db44aaec708e8ce1c/onnxruntime/test/providers/cpu/tensor/quantize_linear_test.cc#L15) do not specify the flag value so it is set to false by default.

The constant initializer flag triggers the optimization, that's why we need it.

Reply:
> They are almost the same, the only difference is that in these new tests the zero point input has the constant initializer flag set to true, this can be seen [here](https://github.com/eralmual/onnxruntime/blob/f6370755fe56e03c07997c238b488047a84a7d0a/onnxruntime/test/providers/dnnl/math/quantize_linear_ops_test.cc#L16), while the [original tests](https://github.com/microsoft/onnxruntime/blob/eb6aa861cfa7295ee9f7145db44aaec708e8ce1c/onnxruntime/test/providers/cpu/tensor/quantize_linear_test.cc#L15) do not specify the flag value so it is set to false by default.
> 
> The constant initializer flag triggers the optimization, that's why we need it.

oh ok. got it. i missed the boolean flag when comparing against the original. thanks!
the question then would be whether these should be added to the original tests or spun off here as a separate set of tests for dnnl only.

Reply:
What I've seen from the original tests is this flag is set to default (false) most of the time, even when that doesn't makes much sense. So I'd rather leave the new ones as dnnl only, so the original tests stay consistent.